reactor-core/src/main/java/reactor/core/publisher/FluxOnAssembly.java

```diff
@@ -378,17 +370,29 @@ final public int requestFusion(int requestedMode) {
 
 		final Throwable fail(Throwable t) {
 			StringBuilder sb = new StringBuilder();
-			fillStacktraceHeader(sb, parent.getClass(), snapshotStack);
 			OnAssemblyException set = null;
-			if (!snapshotStack.isLight()) {
+			final String backtrace;
```

Comment:
to clarify previous review/comments, this could actually be a separate improvement PR:
the `StringBuilder` is only used within the `if/else`, and both paths don't actually depend on any information that is subscriber-specific.
Instead, the whole `String` could be pre-constructed, once, in the `Publisher`.
The downside is that it would eagerly construct the String, even if there are no errors, but at assembly time. @smaldini wdyt?

in the meantime, this PR looks good to me